src/pages/gateway/batching.md

```diff
@@ -0,0 +1,136 @@
+                    "graphql": {
+                        "endpoint": "https://venia.magento.com/graphql"
+                    }
+                }
+            },
+            {
+                "name": "Reviews API",
+                "handler": {
+                    "graphql": {
+                        "endpoint": "<Reviews_API_URL>",
+                        "useGETForQueries":true
+                    }
+                }
+            }
+        ]
+    }
+}
+```
+
+The following query causes multiple calls to the Reviews API:
+
+```graphql
+{
+  productsReviews(sku: ["VT12","VD03"]) {
+    sku
+    reviews {
+      review
+      customer_name
+      rating
+    }
+  }
+}
+```
+
+To make a single network request to each source, modify the mesh configuration to contain the `addtionalTypeDef` and `additionalResolver` described below:
+
+```json
+{
+    "meshConfig": {
+        "sources": [
+            {
+                "name": "Products",
+                "handler": {
+                    "graphql": {
+                        "endpoint": " https://venia.magento.com/graphql"
+                    }
+                }
+            },
+            {
+                "name": "Reviews",
+                "handler": {
+                    "graphql": {
+                        "endpoint": "<Reviews_API_URL>",
+                        "useGETForQueries":true
+                    }
+                }
+            }
+        ],
+        "additionalTypeDefs": "extend type ConfigurableProduct { customer_reviews: productReviewslist} " ,
+        "additionalResolvers" : [
+          {
+            "targetFieldName" : "customer_reviews",
+            "targetTypeName" : "ConfigurableProduct",
+            "sourceName": "Reviews",
+            "sourceTypeName": "Query",
+            "sourceFieldName": "productsReviews",
+            "keysArg": "sku",
+            "keyField": "sku"
+        }],
+        "responseConfig": {
+            "includeHTTPDetails": true
+        }
+    }
+}
+```
+
+The custom resolver extends the type `ConfigurableProdcut` with a new `customer_reviews` field, which allows for the nesting review fields inside of queries against the Venia source. The resolver is composed of the following components:
+
+- The target (`targetTypeName`, `targetFieldName`) - describes the queried field.
+- The source (`sourceName`, `sourceTypeName`, `sourceFieldName`) - describes where the data is resolved for the target field.
+- The keys (`keysArg`, `keyField`):
+  - `keysArg` provides the name of the primary key argument. For this example, the `keysArg` field is the argument name used when sending an array of SKUs to fetch multiple reviews.
+  - `keyField` provides the key value for each item in the batched query. For this example, the `keyField` indicates which Product field provides the SKU value to the review service.
+
+The following query uses the custom resolver to batch the nested data, making only one call to each source:
```

Comment:
If we specify the reviews source specifically then it will be helpful. 
The following query uses the custom resolver to batch the nested data, making only one call to reviews source for different product skus: